### ISSUE TITLE
Add support for jdk18

### ISSUE THREAD

**Issue by mattdailis** [github/codeql#8673]
**Description of the issue**

JDK 18 has just been released, and we'd like to update [our project](https://github.com/NASA-AMMOS/aerie) to use it. However, our CodeQL Github Action is running into some trouble running our (gradle) build:

```
> Task :merlin-sdk:compileJava FAILED
[ODASA Javac] Unsupported Java specification version 18
```

Here is the output of `codeql version --format=json`:
```
  {
    "productName" : "CodeQL",
    "vendor" : "GitHub",
    "version" : "2.8.4",
    "sha" : "5704cd3450dd82e71f1d1df4428f60ec2839ceb4",
    "branches" : [
      "codeql-cli-2.8.4"
    ],
    "copyright" : "Copyright (C) 2019-2022 GitHub, Inc.",
    "unpackedLocation" : "/opt/hostedtoolcache/CodeQL/0.0.0-20220322/x64/codeql",
    "configFileLocation" : "/home/runner/.config/codeql/config",
    "configFileFound" : false
  }
```

Here is the PR in which we are upgrading to Java 18: https://github.com/NASA-AMMOS/aerie/pull/129

Here is the Github Action run of CodeQL that contained the above messages: https://github.com/NASA-AMMOS/aerie/runs/5844585137

Thank you!

**Comment by smowton** [comment]
The Code Scanning distribution upgrade currently in progress will bring preliminary support for JDK18, so this should start working in the next week or so.

**Comment by Marcono1234** [comment]
This preliminary support for JDK 18 has been added in the recently released CodeQL CLI v2.9.0, correct? (also related to #8865)

**Comment by smowton** [comment]
Yes, thanks for the reminder, closing as fixed
